test/TestTransmuteDPOS.spec.js

```diff
@@ -280,6 +286,130 @@ contract('TransmuteDPOS', accounts => {
+    });
+
+    it("should set withdrawBlock for the Delegator's address", async () => {
+      assert.equal(0, await tdpos.withdrawBlocks.call(accounts[2]));
+      await tdpos.unbond({from: accounts[2]});
+      const currentBlockNumber = web3.eth.blockNumber;
+      const unbondingPeriod = (await tdpos.unbondingPeriod.call()).toNumber();
+      const withdrawBlock = await tdpos.withdrawBlocks.call(accounts[2]);
+      assert.equal(currentBlockNumber + unbondingPeriod, withdrawBlock);
+    });
+
+    it('should decrease the totalBondedAmount of the Provider by the unbonded amount', async () => {
+      const totalBondedAmount = (await tdpos.providers(accounts[0]))[5].toNumber();
+      const bondedAmount = (await tdpos.delegators(accounts[3]))[1].toNumber();
+      const newAmount = totalBondedAmount - bondedAmount;
+      await tdpos.unbond({from: accounts[3]});
+      assert.equal(newAmount, (await tdpos.providers([accounts[0]]))[5].toNumber());
+      assert.equal(newAmount, (await tdpos.getProvider(accounts[0]))[0].toNumber());
+    });
+
+    it('should remove the Delegator from the mapping', async () => {
+      let delegator = await tdpos.delegators.call(accounts[4]);
+      assert.notEqual(0,delegator[0]);
+      assert.notEqual(0,delegator[1]);
+      await tdpos.unbond({from: accounts[4]});
+      delegator = await tdpos.delegators.call(accounts[4]);
+      assert.equal(0,delegator[0]);
+      assert.equal(0,delegator[1]);
+    });
+
+    it('should emit the DelegateUnbonded event', async () => {
+      const delegator = await tdpos.delegators.call(accounts[5]);
+      assert.equal(accounts[0], delegator[0]); // [0] is delegateAddress;
+      assert.equal(300, delegator[1]); // [1] is amountBonded;
+      const result = await tdpos.unbond({from: accounts[5]});
+      assert.web3Event(result, {
+        event: 'DelegatorUnbonded',
+        args: {
+          _delegator: accounts[5],
+          _provider: accounts[0],
+          _amount: 300
+        }
+      });
+    });
+  });
+
+  describe('delegatorStatus', () => {
+
+    before(async () => {
+      tdpos = await TransmuteDPOS.new();
+      contractAddress = tdpos.address;
+      for(let i = 0; i < 10; i++) {
+        await tdpos.mint(accounts[i], 1000, {from: accounts[0]});
+      }
+      await tdpos.setMaxNumberOfProviders(PROVIDER_POOL_SIZE);
+      await blockMiner.mineUntilEndOfElectionPeriod(tdpos);
+      await tdpos.initializeRound();
+      await approveBondProvider(22, 10, 1, 25, 1, accounts[0]);
+    });
+
+    it('should return Unbonded if address in not a Delegator', async() => {
+      // Assert that address is not a delegator
+      const delegator = await tdpos.delegators.call(accounts[3]);
+      assert.equal(0,delegator[0]);
+      assert.equal(0,delegator[1]);
+      assert.equal(DELEGATOR_UNBONDED, await tdpos.delegatorStatus(accounts[3]));
+    });
+
+    it('should return Bonded if delegator has called bond()', async() => {
```

Comment:
You capitalized 'Delegator' in all other tests' descriptions, let's keep that consistent.